projects/storefrontlib/src/cms-components/configurator/commons/config-product-title/config-product-title.component.ts

```diff
@@ -20,23 +19,20 @@ export class ConfigProductTitleComponent implements OnInit {
   iconTypes = ICON_TYPE;
 
   constructor(
-    private routingService: RoutingService,
     private configuratorCommonsService: ConfiguratorCommonsService,
     private configRouterExtractorService: ConfigRouterExtractorService,
     private productService: ProductService
   ) {}
 
   ngOnInit(): void {
-    this.product$ = this.configRouterExtractorService
-      .extractRouterData(this.routingService)
-      .pipe(
-        switchMap((routerData) =>
-          this.configuratorCommonsService.getConfiguration(routerData.owner)
-        ),
-        switchMap((configuration) =>
-          this.productService.get(configuration.productCode)
-        )
-      );
+    this.product$ = this.configRouterExtractorService.extractRouterData().pipe(
```

Comment:
Let's move this initialization to the inline property.
The reason for this is:
- `ngOnInit` should be used only when we need to do some processing as soon as Angular passes `@Input()`s (which is not the case in this component.
- It also forces us to maintain this lifecycle hook in case some clients extend this class. And we've already determined that it's not needed in this case.

Reply:
Should we mark this component with `OnPush` strategy, even though it doesn't have `@Input()`s?

Reply:
Didn't know this, added it

Reply:
What is the advantage of doing so?

Reply:
Actually, I just realized that we are not doing it for component that don't have inputs.
There's no advantage, but my idea was to have it specified in case we add inputs in the future. But I guess it might be an over engineering, so let's leave it as is (without any strategy specified).